src/pivot-table/data/extract-headers.ts

```diff
@@ -5,17 +5,18 @@ import getKey from "../components/helpers/get-key";
 
 const extractHeaders = (
   hyperCube: ExtendedHyperCube,
-  rowCount: number,
+  visibleTopDimensionInfo: VisibleDimensionInfo[],
   visibleLeftDimensionInfo: VisibleDimensionInfo[],
 ): (null | HeaderCell)[][] => {
-  const matrix: (null | HeaderCell)[][] = Array(visibleLeftDimensionInfo.length)
+  const rowCount = Math.max(1, visibleTopDimensionInfo.length);
+  const matrix: (null | HeaderCell)[][] = Array(rowCount)
     .fill(null)
-    .map(() => Array.from({ length: rowCount }, () => null));
+    .map(() => Array.from({ length: visibleLeftDimensionInfo.length }, () => null));
 
   visibleLeftDimensionInfo.forEach((qDimensionInfo, colIdx) => {
     if (qDimensionInfo === PSEUDO_DIMENSION_INDEX) {
-      matrix[colIdx][rowCount - 1] = {
-        id: "PSEUDO-DIM",
+      matrix[rowCount - 1][colIdx] = {
+        id: `${PSEUDO_DIMENSION_INDEX}`,
```

Comment:
Consider setting `isPseudoDimension: true` property instead of trying to check against the ID if the cell is a pseudo dimension. Then there should be no need to change the `id` or do the change in `getKey`

Reply:
isPseudoDimension is part of the Cell type. Not the HeaderCell type

This change was to use the same id for pseudo cell instead of introducing a new

Reply:
Changing type from HeaderCell to Cell is better handled by another PR